### PR TITLE
[BEAM-6053] added sparkOptions cacheDisabled

### PR DESCRIPTION
**Pull request by mareksimunek**
Add possibility to SparkOptions to turn off spark RDD caching. There are use cases when its faster to recompute whole RDD rather then serialize, store, deserialize, read from store.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by mareksimunek** [comment]
Run Spark ValidatesRunner

File: runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java [hunk 1]
```diff
@@ -139,11 +139,24 @@ public void setCurrentTransform(AppliedPTransform<?, ?, ?> transform) {
...
-      return true;
+  /**
+   * Cache PCollection if {@link #isCacheDisabled()} flag is false and PCollection is used more then
+   * once in Pipeline or forceCache flag is true.
+   *
+   * @param pvalue
+   * @param forceCache if PCollection is not used more than once
+   * @return if PCollection will be cached
+   */
+  public boolean shouldCache(PValue pvalue, boolean forceCache) {
```
Comment (dmvk) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java]: Can you reason about expected behavior when forceCache is true and the cache is disabled? In which cases we set forceCache to true?

Comment (amitsela) [reply runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java]: it's a user options, if for some reason they want to. @jbonofre might know more, IIRC he brought it up after some folks asked for it in a couple of meetups/conferences.

Comment (jbonofre) [reply runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java]: I don't remember exactly the rational, but AFAIR, users wanted to "force" the caching of RDD (disk storage) even if it's only read once.

Comment (amitsela) [reply runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java]: yeah, I remember the same, but why 🤷‍♂️

**Pull request merged**